### PR TITLE
Generated Payment Card in HotelBookingDetail

### PR DESCRIPTION
**Pull request by xsolano**
We're including inside **HotelBookingDetail** a new object **GeneratedPaymentCard**, which includes a payment card and other information: for now, just a **source** filed which indicated the supplier or origin of this payment card.

We've created also a new object **PaymentCard** in order to use it on a response inside **HoteBookingDetail**. It is a replication of **InputPaymentCard** without the deprecated fields. The **PaymentCardType** is now mandatory (there is not the option to get the type from **cardType** deprecated field).

We're including the **GeneratedPaymentCard** inside **HotelBookingDetail** since we'll be also using it at Booking Step, to show to the customer the payment card that may has been generated at Book step.

### CODE REVIEW

File: admin/core/objects/object_PaymentCard.graphql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+type PaymentCard {
+
+  # Indicates the supported card type
+  type: PaymentCardType!
+
+  # Contains owner's name
+  holder: Holder!
+  
+  # Contains credit card number
+  number: String!
```
Comment (arquio) [inline admin/core/objects/object_PaymentCard.graphql]: # Card numbers are allocated in accordance with ISO/IEC 7812. 
scalar CardNumber

File: admin/core/objects/object_PaymentCard.graphql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  
+  # Contains credit card CVC
+  CVC: String!
```
Comment (arquio) [inline admin/core/objects/object_PaymentCard.graphql]: Use CVC Scalar

File: hotelx/hubgra/objects/object_GeneratedPaymentCard.graphql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Specifies the information of the payment card, and the source of it, that has been generated at Book step
+type GeneratedPaymentCard{
+
+# PaymentCard generated at Book step
+paymentCard: PaymentCard!
+
+# Indicates the supplier or third party, responsible of the payment card's creation
+source: String!
+
```
Comment (arquio) [inline hotelx/hubgra/objects/object_GeneratedPaymentCard.graphql]: supplier:Supplier!

File: admin/core/objects/object_PaymentCard.graphql [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  holder: Holder!
+  
+  # Contains credit card number
+  number: CardNumber!
+  
+  # Contains credit card CVC
+  CVC: CVC!
+  
+  # Contains credit card expiration date
+  expire: ExpireDate!
```
Comment (arquio) [inline admin/core/objects/object_PaymentCard.graphql]: ISO 8601 
Why dont use the DATE or DATETIME object?  @xsolano

Comment (xsolano) [reply admin/core/objects/object_PaymentCard.graphql]: We're following the usual format for expiration dates through most of credit cards (which just requires the year and month). However we just realized that we'll need also the day of expiration. This is why we added an expire date for the createdPaymentCard following in this case the Date format. We've added an active date to the createdPaymentCard too.

This way we'll have the standard PaymentCard with the CVC, CardNumber, Holder and Expire, for all of the products of TravelgateX, and the createdPaymentCard for our HotelX product, with three more parameters (supplier, expire, active), which enable our requeriments.

**Comment by arquio** [comment]
This branch has conflicts that must be resolved @xsolano

**Pull request merged**